### DISCUSSION TITLE
Please someone help out. im 98% lost and 2% sure of things

### DISCUSSION THREAD

**Discussion by webbb82** [twpayne/chezmoi#2979] [General]
ok first let me say I have been googling these issues and honestly I don't know where to start. I cant find my exact issue. I don't know if i should be googling git help or chezmoi help. 
I have chez installed and its running, the first push went well.
- BUT now when I do the daily stuff chezmoi cd, chezmoi diff, ok see the changes i made , chezmoi apply ok, chezmoi push and al that Nothing gets pushed. and my local files get reverted to the files that are in git.  I ran chez doctor and these are some of the warnings

- warning   version              v2.33.4, built at 2023-05-02T17:51:22Z
- warning   source-dir           ~/.local/share/chezmoi is a git working tree (dirty)
- warning   working-tree         ~/.local/share/chezmoi is a git working tree (dirty)
- warning   dest-dir             ~ is a git working tree (dirty)
- info      diff-command         not set
- info      age-command          age not found in $PATH

I have a feeling i saw somewhere there is a cache you need to update but I looked all in the help files and didn't see anything about that. I guess one of my biggest issues is I don't have anyone to just talk to about this to flesh out my confusion. Do I use git like git or do I ignore git and only use it via chezmoi? 

I've looked on youtube for a good video on the finer details of this app but all i can find are surface level info. Maybe that's because that's all there is?  Maybe i'm over thinking this.  I tried using the chezmoi daily list and I get through it but nothing happens. . Thats about all i got. Please dont just say google your questions. I have been, and because Im lost about what to use I dont know what advice to follow

**Comment by twpayne** [comment]
> * BUT now when I do the daily stuff chezmoi cd, chezmoi diff, ok see the changes i made , chezmoi apply ok, chezmoi push and al that Nothing gets pushed.

There is no such command as `chezmoi push`. Please could you provide a terminal transcript that shows exactly what you typed, what you got, and what you expected.

Is it possible that previously you've been using a bare git repo for your dotfiles and are expecting chezmoi to work exactly the same way? chezmoi is different to using a bare git repo.

**Reply by webbb82** [reply]
no i  did follow them exactly. i just said them wrong here. i looked they did upload and they were done via chezmoi because they were changed to chezmois naming convention. Dot_.zshrc Dot_config and so on.
i took some notes the first time i did them so i could do it again

Workflow
```
start chez

chezmoi init
```
```
get into git dir

chezchezmoi cd
```

```
git remote add origin https://github.com/$GITHUB_USERNAME/dotfiles.git
```
```
check whats changed

chezmoi diff
```

```
applies change

chezmoi apply

git push -u origin main
```

So I think i may have found out something since my post do i do chezmoi merge to gwt my files that were modified  out side of chez back in sync? O assumed  i could just edit whatever and it was the job of chez to simply notice a change and upload the new change when i tell it to  push

**Reply by halostatue** [reply]
`chezmoi apply` doesn’t auto-commit the changes, so there may not be a commit to push at this point.

You may need to `git commit -am "my commit message"` *prior* to pushing for it to be reflected on the remote server.

**Reply by webbb82** [reply]
oh ok thank you. thank you fo taking the time. this is just what i needed just a small nudge

**Comment by nandalopes** [comment]
Hi, don't bother with `info` entries about missing programs, unless you're plannig to use then within your templates. Same on `diff` command, chezmoi provides a default value.

Let's check the `warning` entries:

- `warning source-dir ~/.local/share/chezmoi is a git working tree (dirty)`  
  `warning working-tree ~/.local/share/chezmoi is a git working tree (dirty)`

Your chezmoi source dir has uncommited changes, see `git help`.

- `warning dest-dir ~ is a git working tree (dirty)`

Your home directory is a git working tree, maybe you have migrated from another dotfile manager program that create a `~/.git` folder or file.

**Reply by nandalopes** [reply]
About chezmoi workflow:

1. See **chezmoi config file reference**, specially about its variables, for auto push/auto commit git settings.
2. I usually use `re-add` or `add --force` commands for non-template files, and `merge` for templates. But since I use up to three Ubuntu flavors on a daily basis, most of my source state are template files.

My `chezmoi doctor` output (shorten version): [chezmoi-doctor.txt](https://gist.github.com/nandalopes/dfedaf94cd7dbd2deb89719d8d95c8a0)

**Reply by webbb82** [reply]
> About chezmoi workflow:
> 
> 1. See **chezmoi config file reference**, specially about its variables, for auto push/auto commit git settings.
> 2. I usually use `re-add` or `add --force` commands for non-template files, and `merge` for templates. But since I use up to three Ubuntu flavors on a daily basis, most of my source state are template files.
> 
> My `chezmoi doctor` output (shorten version): [chezmoi-doctor.txt](https://gist.github.com/nandalopes/dfedaf94cd7dbd2deb89719d8d95c8a0)

hey thanks for assisting, ya this is the first real repo I did however a while ago try to upload my obsidian notebook and a small ai program I wrote. I think i may  have fixed the dirty dir issues and now when i do chez doc this is the only 

warning   dest-dir             ~ is a git working tree (dirty)

and for my connfig is really only a few lines
[git] 
sourceDir = "/home/bj/dotfiles"
autoCommit = true
autoPush = true

now that im looking at it I dont think its doing any of that auto stuff...maybe I should delete everything and start from scratch with your tips and maybe i should work on a better config
